### PR TITLE
feat: add ApiModel and ApiModelProperty decorator

### PR DESCRIPTION
**Pull request by xrobert35**
Here is "base" working version of apiModel et ApiModelProperty decorator

 We can still manage the model through the old way + the new thanks to a merge on the setDefinition function.

To make the @ApiModel and @ApiModelPropertyWork we need to require every model file before the swagger build.

* I had some problem on vscode for debug cause of the sourcemap (dunno why yet)
* You/we should create an editorconfig file at the root of the project to manage code formatting http://editorconfig.org/

### CODE REVIEW

File: src/lib/swagger-express-ts/swagger.service.ts [hunk 1]
```diff
@@ -114,8 +117,8 @@ export class SwaggerService {
         this.data.host = host;
     }
 
-    public setDefinitions( definitions: {[key: string]: ISwaggerDefinition} ): void {
-        this.data.definitions = definitions;
+    public setDefinitions(definitions: { [key: string]: ISwaggerDefinition }): void {
+        this.data.definitions = _.merge(this.data.definitions, definitions);
```
Comment (xrobert35) [inline src/lib/swagger-express-ts/swagger.service.ts]: merge the "old style model" with the decorator model

File: src/main.ts [hunk 2]
```diff
@@ -6,7 +6,9 @@ import { interfaces, InversifyExpressServer, TYPE } from "inversify-express-util
 import { VersionController } from "./version/version.controller";
 import * as swagger from "./lib/swagger-express-ts";
 import { SwaggerDefinitionConstant } from "./lib/swagger-express-ts";
-const config = require( "../config.json" );
+const config = require("../config.json");
+
+import './version/version.model';
```
Comment (xrobert35) [inline src/main.ts]: loading the model before build

File: src/lib/swagger-express-ts/swagger.service.ts [hunk 3]
```diff
@@ -42,9 +46,9 @@ interface IController {
 }
 
 export class SwaggerService {
+
     private static instance: SwaggerService;
-    private controllerMap: any = [];
-    private definitionsMap: {[key: string]: ISwaggerDefinition} = {};
```
Comment (xrobert35) [inline src/lib/swagger-express-ts/swagger.service.ts]: unused var

File: src/lib/swagger-express-ts/swagger.service.ts [hunk 4]
```diff
@@ -42,9 +46,9 @@ interface IController {
...
+    private controllerMap: IController[] = [];
```
Comment (xrobert35) [inline src/lib/swagger-express-ts/swagger.service.ts]: change type from any to > IController[]

File: src/lib/swagger-express-ts/api-model-property.decorator.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  required? : boolean
+  format?: string;
+  type?: string;
+  description?: string;
+  enum?: string[];
+}
+
+export function ApiModelProperty(args?: IApiModelPropertyArgs): PropertyDecorator {
+  return function (target: any, propertyKey: string | symbol) {
+    const propertyType = Reflect.getMetadata("design:type", target, propertyKey).name.toLowerCase();
```
Comment (xrobert35) [inline src/lib/swagger-express-ts/api-model-property.decorator.ts]: get the property type through reflect-metadata,  I need to test more case

**Comment by daniloavilez** [comment]
Nice man, it was really fast!! I think it will improve a lot this project!

**Comment by xrobert35** [comment]
@olivierlsc do you see some enhancement to add ? 
what are we waiting ? :D

**Comment by olivierlsc** [comment]
@xrobert35 this enhancement has been added to the "develop" branch for next tag 1.0.0-beta2. Good job! Thanks for your help

**Pull request merged**